### PR TITLE
GCP EventThreatDetection parser

### PR DESCRIPTION
**Pull request by adrianosela**
## ETD Finding Parser. 

During the beta, ETD Findings will be available through stackdriver logs -- they are also available through Cloud SCC in GCP.

### In this PR:
* Event Threat Detection Finding JSON model
* Event Threat Detection Finding parser
* Tests for ETD parser with and without stackdriver encapsulation

### CODE REVIEW

File: src/main/java/com/mozilla/secops/parser/ETDBeta.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import com.google.api.services.logging.v2.model.LogEntry;
+import com.mozilla.secops.parser.models.etd.EventThreatDetectionFinding;
+import java.io.IOException;
+import java.io.Serializable;
+
+/** Payload parser for GCP ETD Finding data */
+public class ETDBeta extends PayloadBase implements Serializable {
+
+  private static final long serialVersionUID = 1L;
+  public static final String STACKDRIVER_LOG_RESOURCE_TYPE = "threat_detector";
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/ETDBeta.java]: Add javadoc if we are making this public

File: src/main/java/com/mozilla/secops/parser/ETDBeta.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  /** Construct matcher object. */
+  public ETDBeta() {}
+
+  /**
+   * Construct parser object.
+   *
+   * @param input Input string
+   */
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/ETDBeta.java]: Missing other params in doc comments here

File: src/main/java/com/mozilla/secops/parser/Parser.java [hunk 3]
```diff
@@ -237,6 +237,15 @@ private String stripStackdriverEncapsulation(Event e, String input, ParserState
       // We were able to deserialize the LogEntry so store it as a hint in the state
       state.setLogEntryHint(entry);
 
+      // If the JSON Payload is an ETD Finding, return only the payload.
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/Parser.java]: Let's take this section out for now, we don't want payload specific edge cases in here

Was this needed or is this more an efficiency thing?

**Comment by adrianosela** [comment]
@ameihm0912 The TODO's are things related to the "TODO" section in the PR description. The doc that was given to us by the big G included some invalid JSON for some fields, but this doesnt stop us from parsing and categorizing the findings, as those field are just additional metadata that wont really have an impact on whether we alert or not for a given finding.

Will address other comments and get this ready.

**Comment by adrianosela** [comment]
The TODO's will be fixed/updated whenever we can get some actual data out of ETD to confirm the format of those fields.

File: src/main/java/com/mozilla/secops/parser/Parser.java (hunk 3 above)
Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/Parser.java]: This was uniquely an efficiency thing.

File: src/main/java/com/mozilla/secops/parser/ETDBeta.java (hunk 1 above)
Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/ETDBeta.java]: ✅

File: src/main/java/com/mozilla/secops/parser/ETDBeta.java (hunk 2 above)
Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/ETDBeta.java]: ✅

File: src/main/java/com/mozilla/secops/parser/Parser.java (hunk 3 above)
Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/Parser.java]: ✅

File: src/main/java/com/mozilla/secops/parser/ETDBeta.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+   *
+   * @param input Input string
+   * @param e Parent {@link Event}
+   * @param s State {@link ParserState}
+   */
+  public ETDBeta(String input, Event e, ParserState s) {
+
+    // try to parse from a stack driver log
+    try {
+      LogEntry entry = jf.createJsonParser(input).parse(LogEntry.class);
```
Comment (ameihm0912) [inline src/main/java/com/mozilla/secops/parser/ETDBeta.java]: You should be able to reuse the existing state here

See also https://github.com/mozilla-services/foxsec-pipeline/blob/f1963d94eec8b0bb547e3e0dfa2e939c2a6cfa32/src/main/java/com/mozilla/secops/parser/GLB.java#L75-L96

Comment (adrianosela) [reply src/main/java/com/mozilla/secops/parser/ETDBeta.java]: ✅

**Comment by adrianosela** [comment]
@ameihm0912 big G team gave me a better example output and I completed all the items marked "TODO" in this PR. ✅

**Pull request merged**